src/metal.cs

```diff
@@ -444,9 +447,13 @@ partial interface MTLComputeCommandEncoder : MTLCommandEncoder {
 		[Export ("dispatchThreadgroupsWithIndirectBuffer:indirectBufferOffset:threadsPerThreadgroup:")]
 		void DispatchThreadgroups (IMTLBuffer indirectBuffer, nuint indirectBufferOffset, MTLSize threadsPerThreadgroup);
 
+#if XAMCORE_4_0
 		[Abstract]
+#else
 		[Export ("setBuffers:offsets:withRange:")]
-		void SetBuffers (IMTLBuffer [] buffers, IntPtr offsets, NSRange range);
+		void SetBuffers (IntPtr buffers, IntPtr offsets, NSRange range);
+#endif
+
```

Comment:
This isn't quite right for XAMCORE_4_0, it'll end up as just the `[Abstract]` attribute and nothing else.